web/packages/teleterm/src/services/tshd/cloneableClient.ts

```diff
@@ -254,9 +254,45 @@ export type TshdRpcError = Pick<
   isResolvableWithRelogin: boolean;
 };
 
-/** Checks if the given value is a `TshdRpcError`. */
-export function isTshdRpcError(error: unknown): error is TshdRpcError {
-  return error['name'] === 'TshdRpcError';
+/**
+ * Subset of gRPC status codes.
+ * It doesn't contain UNAUTHENTICATED and ABORTED values, which could be
+ * mistakenly used as PERMISSION_DENIED and CANCELLED.
```

Comment:
I appreciate the thought, but I'm not totally sold on hiding `ABORTED`. AFAIK, [`CANCELLED` is typically used for cancellation](https://grpc.io/docs/guides/cancellation/) by the client.

`ABORTED` is meant to be used on the server side:

> Service implementors can use the following guidelines to decide between FAILED_PRECONDITION, ABORTED, and UNAVAILABLE: (a) Use UNAVAILABLE if the client can retry just the failing call. (b) Use ABORTED if the client should retry at a higher level (e.g., when a client-specified test-and-set fails, indicating the client should restart a read-modify-write sequence). (c) Use FAILED_PRECONDITION if the client should not retry until the system state has been explicitly fixed.

We actually do use `ABORTED` and it's when tshd sends `PromptMFA` to the Electron app. If the user closes the modal, the Electron app responds with `ABORTED` (see below).

Overall I feel like we should keep `ABORTED` to avoid the inverse – encourage people to use `CANCELLED` where `ABORTED` would be better suited.

In the context of a client `ABORTED` seems important. A client will have little need to check for `CANCELLED` since cancellation will typically be done on client side, by e.g. aborting a signal. On the other hand, there might be scenarios where a client needs to know if the server `ABORTED` a call. 

https://github.com/gravitational/teleport/blob/ad015469fb9c4d59584cbbd5be1517393cf38837/web/packages/teleterm/src/ui/tshdEvents.ts#L76-L85

https://github.com/gravitational/teleport/blob/ad015469fb9c4d59584cbbd5be1517393cf38837/web/packages/teleterm/src/services/tshdEvents/index.ts#L202-L211

https://github.com/gravitational/teleport/blob/ad015469fb9c4d59584cbbd5be1517393cf38837/lib/teleterm/daemon/mfaprompt.go#L93-L97

Reply:
That makes sense. I will also add `UNAUTHENTICATED`, I think there is no justification for omitting a single value.